includes/wf_crm_webform_postprocess.inc

```diff
@@ -2035,7 +2039,29 @@ class wf_crm_webform_postprocess extends wf_crm_webform_base {
     $params['financial_type_id'] = $this->contribution_page['financial_type_id'];
     $params['currency'] = $params['currencyID'] = $this->contribution_page['currency'];
     $params['skipRecentView'] = $params['skipLineItem'] = 1;
-    $params['contact_id'] = $this->ent['contact'][1]['id'];
+    // Get contribution contact id.
+    $contribution_contact_id = $params['contribution_contact_id'];
+    // Set default contact 1 if contribution contact is empty.
+    // This is specifically added for user select option.
+    // If user do not choose any option from contribution contact field,
+    // then assign default contact 1 as contribution contact.
+    if (empty($contribution_contact_id)) {
+      $contribution_contact_cid = $this->ent['contact'][1]['id'];
+    }
+    // Pay later option, where payment processor is not selected.
+    elseif (empty($params['payment_processor_id'])) {
+      $contribution_contact_cid = $contribution_contact_id;
+    }
+    else {
+      $contribution_contact_cid = $this->ent['contact'][$contribution_contact_id]['id'];
+    }
+    // For new contact cid is empty.
+    // Create new contact to get the cid.
+    if (!$contribution_contact_cid) {
+      $contact = $this->data['contact'][$contribution_contact_id];
+      $this->ent['contact'][$contribution_contact_id]['id'] = $contribution_contact_cid = $this->createContact($contact);
```

Comment:
@viraj-compuco 

The duplicate contact that you are talking about above seems to be a ramification of you adding this new line, which is not a line that makes sense to be added here because this function purpose is to set the contribution parameters not to create contacts. Also all contacts at this point should already be created by the preSave() function so I am not sure why you have to do this here.  So unless you have a better solution than this  I will have to deploy this PR on my local and check few things before I can tell you if I can approve it or not.

Reply:
@omarabuhussein Lets have short call to discuss this changes

Reply:
@jamienovick Quick question
We have to fill Billing Name and Address on the payment page, currently it is of Contact 1 only.
Now with new implementation, we are assigning the contribution to other contacts also, Can we change the Billing Name and Address for the selected contribution contact or keep it as it is for contact 1?

cc @omarabuhussein

Reply:
The billing name and address should be attached to the contact who makes the payment. Thanks

Reply:
I actually don't think we care who the contact who submits the form is, even if its a payment processor. Let me run a scenario for you:

Eg: John Smith comes to form but wants to pay for membership for his organisation:

- Individual is Contact 1
- Org Contact is Contact 2
- Membership 1 is attached to Contact 2
- Contribution contact is Contact 2 (i.e. the Organisation)

We want the contribution (invoice) to be connected to contact 2. 
If paying by Stripe, John will enter his credit card details, but he is still paying for Org Contact 
The contribution will show the "billing" information as a separate non editable block at the bottom of the page:

![image](https://github.com/compucorp/webform_civicrm/assets/2102899/a49cd767-2270-4954-80cb-58a2fb126d6b)

I don't think there is any issue with this being a Stripe transaction or Pay later.

Basically we should think of the contribution as the "invoice" and the payment as a separate entity. The invoice would be addressed to contact 2, but the payment was made by contact 1. What would probably be preferable would be that the billing address information was a property of the payment, and not the contribution - I'm not sure what the data structure is for this but at least CiviCRM stores the details of who made the payment.